plugin/v2/plugin.go

```diff
@@ -149,13 +149,59 @@ next:
 				} else if !ok {
 					return fmt.Errorf("%q is not settable", s.prettyName())
 				}
-				// is it, so lets update the config in memory
+				// it is, so lets update the config in memory
 				updateConfigEnv(&p.PluginObj.Config.Env, &s)
 				continue next
 			}
 		}
 
-		//TODO: check devices, mount and args
+		// range over all the mounts in the manifest
+		for _, mount := range p.PluginObj.Manifest.Mounts {
+			// found the mount in the manifest
+			if mount.Name == s.name {
+				// is it settable ?
+				if ok, err := s.isSettable(allowedSettableFieldsMounts, mount.Settable); err != nil {
+					return err
+				} else if !ok {
+					return fmt.Errorf("%q is not settable", s.prettyName())
+				}
+
+				// it is, so lets update the config in memory
+				updateConfigMounts(p.PluginObj.Config.Mounts, &s)
```

Comment:
Dont need to send the entire Mounts list to the function. By now, we already know which mount `m` in ` p.PluginObj.Manifest.Mounts`  has the match. So pass `m` as an arg and simply update that, instead of performing yet another redundant loop in updateConfigMounts.